python/ray/tests/test_task_metrics.py

```diff
@@ -456,19 +456,22 @@ def f(x):
 ray.get([f.remote(x) for x in buf])"""
 
     proc = run_string_as_driver_nonblocking(driver)
-    expected = {
-        "RUNNING": 2.0,
-        "PENDING_ARGS_FETCH": 7.0,
-        "PENDING_OBJ_STORE_MEM_AVAIL": 91.0,
-    }
+
+    # This test is non-deterministic since pull bundles can sometimes end up fallback
```

Comment:
Hmm actually why does this happen? Looking at the code, it's not obvious why we need to fallback allocate (fine with this, just wondering if we need to track this issue).

Reply:
I believe ReceiveChunk() has fallback allocate always enabled, but I'm not entirely sure about the reason. I don't think there is an issue here beyond nondeterminism; the throttling does happen properly sans these few extra allocations.